### ISSUE TITLE
cd auto-completion quoting issue on Windows hosts

### ISSUE THREAD

**Issue by TacoVox** [elves/elvish#897]
Let me start with thanking you for this wonderful project. I really enjoy it! And it works like a charm on my macOS machine.

---

However, I am forced to use MS Windows 10 at work. As I want to use `elvish` for my daily work, I encountered an issue when using the auto-completion on the press of `TAB`.

To visualize the problem, let me show you how a path builds up using the feature:
```sh
~ > cd
[TAB]
~ > cd go'\'
[TAB]
~ > cd 'go\pkg''\'
```

Needless to say, the user would expect that to look like:
```sh
~ > cd 'go\pkg\'
```

I believe to have identified some lines that are partly causing the issue:

`pkg/edit/complete/raw_item.go:41+`
```go
func (c ComplexItem) Cook(q parse.PrimaryType) completion.Item {
	quoted, _ := parse.QuoteAs(c.Stem, q)

	return completion.Item{
		ToInsert:  quoted + c.CodeSuffix, // <- a simple concatenation
		ToShow:    c.Stem + c.DisplaySuffix,
		ShowStyle: c.DisplayStyle,
	}
}
```

Due to the concatenation, we have these multiple single cuoted expressions after one another.

As I see it, one could argue that whenever there is such a case, all single-quoted expressions should  be combined into a single one. Howevever, as I just started looking at the code, I have not yet a complete overview and this may be a bad idea.

@xiaq what do you think? How could this issue be effectively avoided? I would be glad to help out.

---

Thanks for your work. Much appreciated!

**Comment by TacoVox** [comment]
To follow up on my issue:

What I mean is an additional function in the `parse` package that is capable of combining `n` quoted expressions into a single one. Of course it would need to feature logic to check, if we need to quote with single quotes or double quotes.

Is there any case where such a function would cause harm? Otherwise the marked line above could look something along the lines of:
```go
func (c ComplexItem) Cook(q parse.PrimaryType) completion.Item {
	[...]
	return completion.Item{
		[...]
		ToInsert: parse.CombineQuotes(quoted, c.CodeSuffix),
		[...]
	}
}
```

**Comment by xiaq** [comment]
Rather than combining quotes, a simpler approach is to make it possible to attach a code suffix before quoting.

**Comment by xiaq** [comment]
Hmm, an even simpler way is to just add the path separator to the completion item itself without relying on CodeSuffix. This has the side effect of showing the path separator in the listing, but that is what every other shell does anyway (I tested bash, fish and zsh and they all do this).

**Comment by TacoVox** [comment]
That does sound like a nice approach. Shall I look into it? Would be glad to assisst.

**Comment by xiaq** [comment]
@TacoVox Feel free to hack the code and open a PR!

**Comment by xiaq** [comment]
Fixed in #903.
